src/edu/ucsb/nceas/metacat/admin/PropertiesAdmin.java

```diff
@@ -88,29 +88,119 @@ public void configureProperties(HttpServletRequest request,
 
                 // Attempt to discover the following properties.  These will show
                 // up in the configuration fields if nothing else is provided.
-                PropertyService.setPropertyNoPersist("application.context",
-                        ServiceService.getRealApplicationContext());
-                PropertyService.setPropertyNoPersist("server.name", SystemUtil
-                        .discoverServerName(request));
-                PropertyService.setPropertyNoPersist("server.httpPort", SystemUtil
-                        .discoverServerPort(request));
-                PropertyService.setPropertyNoPersist("server.httpSSLPort",
-                        SystemUtil.discoverServerSSLPort(request));
-                PropertyService.setPropertyNoPersist("application.deployDir",
-                    SystemUtil.discoverDeployDir(request));
-                PropertyService.setPropertyNoPersist("application.datafilepath",
-                        externalDir + FileUtil.getFS() + "data");
-                PropertyService.setPropertyNoPersist("application.inlinedatafilepath",
-                        externalDir + FileUtil.getFS() + "inline-data");
-                PropertyService.setPropertyNoPersist("application.documentfilepath",
-                        externalDir + FileUtil.getFS() + "documents");
-                PropertyService.setPropertyNoPersist("application.tempDir",
-                        externalDir + FileUtil.getFS() + "temporary");
-                PropertyService.setPropertyNoPersist("replication.logdir",
-                        externalDir + FileUtil.getFS() + "logs");
-                PropertyService.setPropertyNoPersist("solr.homeDir",
-                        externalDir + FileUtil.getFS() + "solr-home");
-
+                String applicationContext = null;
+                try {
+                    applicationContext = PropertyService.getProperty("application.context");
+                } catch (PropertyNotFoundException ee) {
+                    applicationContext = null;
+                }
+                if (applicationContext == null || applicationContext.trim().equals("")) {
+                    PropertyService.setPropertyNoPersist("application.context",
+                            ServiceService.getRealApplicationContext());
+                }
```

Comment:
(applies to lines 91 thru' 200)

Instead of repeating this pattern multiple times, can we refactor it out into a utility method? That way, this code block would be much more readable and less error-prone for future editing. Here's an example:

1. First define a method, e.g.:
```
            private boolean isNotSet(String propertyKey) {
                String property = null;
                try {
                    property = PropertyService.getProperty(propertyKey);
                } catch (PropertyNotFoundException ee) {
                    property = null;
                }
                return !(property == null || property.trim().equals(""));
            }
```

2. then instead of having this for each property:
```
                String applicationContext = null;
                try {
                    applicationContext = PropertyService.getProperty("application.context");
                } catch (PropertyNotFoundException ee) {
                    applicationContext = null;
                }
                if (applicationContext == null || applicationContext.trim().equals("")) {
                    PropertyService.setPropertyNoPersist("application.context",
                            ServiceService.getRealApplicationContext());
                }
```
...we can simplify to:
```
                if (isNotSet("application.context") {
                    PropertyService.setPropertyNoPersist("application.context",
                            ServiceService.getRealApplicationContext());
                }
```

Reply:
Matthew, do you think in the `isNotSet` method, the return statement should be `return (property == null || property.trim().equals(""));` rather than `return !(property == null || property.trim().equals(""));`?

Reply:
you are correct - sorry about that!